### DISCUSSION TITLE
How to do rotation for /var/log/coraza/audit/audit.log ?

### DISCUSSION THREAD

**Discussion by uniuuu** [corazawaf/coraza#486] [Q&A]
How to rotate audit log? /var/log/coraza/audit/audit.log
It grows and for one single website it's making 6GB in one month.

**Comment by jcchavezs** [comment]
A quick google search on how modsecurity handles this is mainly by not
handling it and let other tools (Apache or ngnx) to deal with it.
Personally I don't think coraza should support log rotation mechanism but I
agree we should describe how to achieve it (maybe an example) and expose
right interfaces for that. For example, what happens if file changes while
the destination file is open? Can it detect a change and reopen the file
accordingly?

**Reply by jptosso** [reply]
I'm not sure if logrotate is going to work. Maybe we should add a file handler to detect rotation and move the writer to 0,0.
Let's run some tests and add documentation.

**Reply by prologic** [reply]
Just came across this, and using `logrotate` does not work. It requires a restart of Caddy. It looks like the logfile is not reopened when it gets renamed and re-created.

**Reply by ckuethe** [reply]
I too would like to see some log rotation, especially with Caddy as the host. The same sort of logic that `tail -F` uses to watch for a source file being replaced could probably watch for a destination file name going away to trigger a reopen, eg. https://github.com/coreutils/coreutils/blob/master/src/tail.c#L972
